front/app/modules/commercial/moderation/admin/containers/index.tsx

```diff
@@ -253,83 +236,104 @@ const Moderation = memo<Props & InjectedIntlProps>(({ className, intl }) => {
     },
   ]);
 
-  const handleOnSelectAll = useCallback(
-    (_event: React.ChangeEvent) => {
-      if (!processing) {
-        if (!isNilOrError(moderations)) {
-          setSelectedModerations(
-            selectedModerations.length < moderations.length ? moderations : []
-          );
-        }
+  const {
+    list: moderations,
+    pageSize,
+    moderationStatus,
+    currentPage,
+    lastPage,
+    onModerationStatusChange,
+    onPageNumberChange,
+    onPageSizeChange,
+    onModeratableTypesChange,
+    onProjectIdsChange,
+    onSearchTermChange,
+    onIsFlaggedChange,
+  } = useModerations({
+    pageSize: selectedPageSize,
+    moderationStatus: 'unread',
+  });
+  const moderationsWithActiveFlag = useModerations({
+    isFlagged: true,
+  });
+
+  const handleOnSelectAll = (_event: React.ChangeEvent) => {
+    if (!processing) {
+      if (!isNilOrError(moderations)) {
+        setSelectedModerations(
+          selectedModerations.length < moderations.length ? moderations : []
+        );
       }
-    },
-    [moderations, selectedModerations, processing]
-  );
+    }
+  };
 
-  const handleOnTabChange = useCallback(
-    (tabName: TActivityTabName) => {
-      setSelectedTab(tabName);
+  const handleOnTabChange = (tabName: TActivityTabName) => {
+    setSelectedTab(tabName);
+    trackEventByName(tracks.tabClicked, {
+      tabName,
+    });
+  };
 
-      if (tabName === 'read' || tabName === 'unread') {
-        onIsFlaggedChange(false);
-        onModerationStatusChange(tabName);
-      }
+  useEffect(() => {
+    if (selectedTab === 'read' || selectedTab === 'unread') {
+      onIsFlaggedChange(false);
+      onModerationStatusChange(selectedTab);
+    }
 
-      // OS: how to?
-      if (tabName === 'warnings') {
-        onIsFlaggedChange(true);
-        onModerationStatusChange(null);
-      }
+    // OS: how to?
+    if (selectedTab === 'warnings') {
+      onIsFlaggedChange(true);
+      onModerationStatusChange(null);
+    }
+  }, [selectedTab]);
```

Comment:
I see missing dependencies in these useEffects - `onIsFlaggedChange`, `onModerationStatusChange`.

Reply:
Added all the missing ones. Thanks for noticing!